src/repr/src/row/collection.rs

```diff
@@ -511,37 +629,36 @@ mod tests {
     fn test_sorted_iter_offset() {
         let a = Row::pack_slice(&[Datum::String("hello world")]);
         let b = Row::pack_slice(&[Datum::UInt32(42)]);
-        let col = RowCollection::new(&[
-            (a.clone(), NonZeroUsize::new(3).unwrap()),
-            (b.clone(), NonZeroUsize::new(2).unwrap()),
-        ]);
-        let col = col.sorted_view(|a, b| a.cmp(b));
+        let col = vec![
+            RowCollection::new(&[(a.clone(), NonZeroUsize::new(3).unwrap())]),
+            RowCollection::new(&[(b.clone(), NonZeroUsize::new(2).unwrap())]),
+        ];
 
         // Test with a reasonable offset that does not span rows.
-        let mut iter = col.into_row_iter().apply_offset(1);
+        let mut iter = SortedRowCollections::new(col.clone())
+            .into_row_iter()
+            .apply_offset(1);
         assert_eq!(iter.next(), Some(b.borrow()));
         assert_eq!(iter.next(), Some(a.borrow()));
         assert_eq!(iter.next(), Some(a.borrow()));
         assert_eq!(iter.next(), Some(a.borrow()));
         assert_eq!(iter.next(), None);
         assert_eq!(iter.next(), None);
 
-        let col = iter.into_inner();
```

Comment:
Removed this function as the collection removes entries at the end of an iteration.

To keep the logic simple, simply recreating the iterator from scratch every time.